ext/php8/handlers_curl.c

```diff
@@ -10,6 +10,11 @@
 #include "logging.h"
 #include "span.h"
 
+#if !(defined(HAVE_CURL) && !defined(COMPILE_DL_CURL))
+__attribute__((weak)) zend_class_entry *curl_ce = NULL;
+__attribute__((weak)) zend_class_entry *curl_multi_ce = NULL;
+#endif
```

Comment:
If these weak symbols don't get linked in, what's the behavior?

Reply:
These symbols should always be linked in when curl isn't compiled with `--with-curl` or when `--with-curl=shared` but curl isn't loaded via INI. (In those cases, none of the codepaths in this file are reached.) I'll have some tests in CI to validate this behavior by EOD tomorrow.

Reply:
Here's how [ext/uv fixed a similar issue](https://github.com/amphp/ext-uv/commit/f7d052a48b8a63ae090e069d42e4278b33e69484) so I think we also need to make use of `DL_FETCH_SYMBOL` in MINIT when curl is loaded as a shared lib. I'll make sure the tests hit that scenario and update the fix.

Reply:
> These symbols should always be linked
Turns out on some platforms the symbols need to be fetched with `dlsym()`. I've added this to MINIT.